### PR TITLE
Aka bugfix

### PR DESCRIPTION
**Pull request by Alihan26**
Sopra-fs24-group07/sopra-fs24-group-07-server#150
-SessionService.java

# Pull Requests Checklist

Assignee:

- \[x\] code commented?
- \[x\] tests added / passed?
- \[x\] are tests covering all cases?
- \[ \] documentation added / updated?
- \[x\] pre-commit checks passed?
- \[x\] pipelines passed?

Reviewer:

- \[x\] correct branch?
- \[x\] are there any code smells?
- \[x\] did pipeline pass?
- \[x\] is the code well documented? -> in code ok
- \[x\] are the tests passing and covering all cases?

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java
Comment (5v3nn) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java:1]: `get sessions` would still result in an > 24h active session and throw the users in the session. this is what we are trying to avoid.

File: src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java
Comment (5v3nn) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java:139]: helper methods should be private or protected (expose minimal interface!)

**Review by 5v3nn — CHANGES_REQUESTED** [review]

File: src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java
Comment (Alihan26) [reply src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java:139]: I changed it to Privat @5v3nn

File: src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java
Comment (Alihan26) [reply src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java:1]: It works now, I would appreciate your review @5v3nn

**Comment by Alihan26** [comment]
The quality gate has also passed @5v3nn

File: src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java
Comment (5v3nn) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java:135]: every minute is too expensive. make every hour

File: src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java
Comment (5v3nn) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java:144]: too expensive. rather write a new repo method to get all sessions which have an empty end date field

File: src/test/java/ch/uzh/ifi/hase/soprafs24/service/SessionServiceIntegrationTest.java
Comment (5v3nn) [inline src/test/java/ch/uzh/ifi/hase/soprafs24/service/SessionServiceIntegrationTest.java:366]: seems like a flaky test. maek minus days 1 and minus a few hours, just to be save. you never know with time tests

File: src/test/java/ch/uzh/ifi/hase/soprafs24/service/SessionServiceTest.java
Comment (5v3nn) [inline src/test/java/ch/uzh/ifi/hase/soprafs24/service/SessionServiceTest.java:296]: same as above

**Review by 5v3nn — CHANGES_REQUESTED** [review]

**Comment by 5v3nn** [comment]
documentation still not added. this is crucial if we make a scheduled task!

File: src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java
Comment (Alihan26) [reply src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java]: Changed it to:
@Scheduled(fixedRate = 3600000)
@5v3nn

File: src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java
Comment (Alihan26) [reply src/main/java/ch/uzh/ifi/hase/soprafs24/service/SessionService.java]: made a new repo method:
findByEndDateTimeIsNull()
@5v3nn

File: src/test/java/ch/uzh/ifi/hase/soprafs24/service/SessionServiceIntegrationTest.java
Comment (Alihan26) [reply src/test/java/ch/uzh/ifi/hase/soprafs24/service/SessionServiceIntegrationTest.java]: you're correct

File: src/test/java/ch/uzh/ifi/hase/soprafs24/service/SessionServiceTest.java
Comment (Alihan26) [reply src/test/java/ch/uzh/ifi/hase/soprafs24/service/SessionServiceTest.java]: handled

**Comment by Alihan26** [comment]
> @Alihan26 documentation still not added. this is crucial if we make a scheduled task!

before the merge, the final implementation to the document will be added. Awaiting merge approval

**Review by 5v3nn — APPROVED** [review]
very nice. now we have something that will have a low runtime and works guaranteed. thank you

**Pull request merged**
